src/box/xlog.c

```diff
@@ -696,6 +698,34 @@ xdir_collect_garbage(struct xdir *dir, int64_t signature, unsigned flags)
 	}
 }
 
+int
+xdir_remove_file_by_vclock(struct xdir *dir, struct vclock *to_remove)
+{
+	struct vclock *find = vclockset_psearch(&dir->index, to_remove);
+	/*
+	 * See vclockset_match(): vclockset may return 0 for incomparable keys
```

Comment:
But you could call the match, and then check its result instead of doing the manual search.

Reply:
Indeed, let's reuse vclockset_match(). Updates on the branch